packages/kbn-observable/src/lib/track_subscriptions.ts

```diff
@@ -0,0 +1,35 @@
+import { Observable, Subscription } from '../observable';
+
+/**
+ * Test helper used to track the first `subscribe` on each of the input
+ * observables.
+ *
+ * If Jest was able to track return values from `spyOn`s, this would be much
+ * simpler. See https://github.com/facebook/jest/issues/3821 for the Jest issue
+ * to add this feature.
+ */
+export function trackSubscriptions<T>(...observables: Observable<T>[]) {
```

Comment:
How would you feel about stubbing `Observable.subscribe`, or hooking into `SubscriptionImpl` somehow, so that even if a test fails to track all observables we can ensure that all Subscriptions are closed when `ensureSubscriptionsAreClosed` is called?

Reply:
I tried a couple ways, but didn't find a smooth solution to it. Not a huge fan of this impl, though, so open to other implementations if you want to give it a try.